roxctl/common/tls.go

```diff
@@ -48,26 +48,33 @@ func (v *insecureVerifierWithWarning) VerifyPeerCertificate(leaf *x509.Certifica
 }
 
 // ConnectNames returns the endpoint and (SNI) server name given by the
-// --endpoint and --server-name flags respectively. If no server name is given,
-// an appropriate name is derived from the given endpoint.
-func ConnectNames() (string, string, error) {
-	endpoint, _, err := flags.EndpointAndPlaintextSetting()
+// --endpoint and --server-name flags respectively and information about plaintext.
+// If no server name is given, an appropriate name is derived from the given endpoint.
+func ConnectNames() (string, string, bool, error) {
+	endpoint, usePlaintext, err := flags.EndpointAndPlaintextSetting()
 	if err != nil {
-		return "", "", errors.Wrap(err, "could not get endpoint")
+		return "", "", false, errors.Wrap(err, "could not get endpoint")
 	}
-	serverName := flags.ServerName()
-	if serverName == "" {
-		var err error
-		serverName, _, _, err = netutil.ParseEndpoint(endpoint)
-		if err != nil {
-			return "", "", errors.Wrap(err, "could not parse endpoint")
-		}
+	serverName, err := getServerName(endpoint)
+	if err != nil {
+		return "", "", false, errors.Wrap(err, "could not get server name")
+	}
+	return endpoint, serverName, usePlaintext, nil
+}
+
+func getServerName(endpoint string) (string, error) {
+	if serverName := flags.ServerName(); serverName != "" {
+		return serverName, nil
+	}
+	serverName, _, _, err := netutil.ParseEndpoint(endpoint)
+	if err != nil {
+		return "", errors.Wrap(err, "could not parse endpoint")
```

Comment:
Well, this is not what I meant, I'm sorry for being unclear. Please revert the refactoring, as it doesn't make any sense now. I'll come up with a follow-up PR, if it will worth it.